css/datagrid/datagrid.css

```diff
@@ -691,6 +692,41 @@ th.highcharts-datagrid-column-sortable > .highcharts-datagrid-column-sortable-ic
     text-align: left;
 }
 
+.highcharts-datagrid-loading-wrapper {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    background: var(--highcharts-neutral-color-0);
+    opacity: 0.5;
+    gap: 10px;
+}
+
+.highcharts-datagrid-loading-wrapper .highcharts-datagrid-loading-message {
+    color: var(--highcharts-neutral-color-100);
+}
+
+.highcharts-datagrid-loading-wrapper .highcharts-datagrid-spinner {
+    border: 4px solid var(--highcharts-neutral-color-60);
+    border-top: 4px solid var(--highcharts-neutral-color-100);
```

Comment:
It would be good to parametrize styling by variables. I mean border-width and separate color.

Could be included in the new theming PR.